internal/metrics/data_transfer.go

```diff
@@ -0,0 +1,88 @@
+package metrics
+
+import (
+	"fmt"
+	"reflect"
+	"sync/atomic"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+
+	"github.com/project-flotta/flotta-device-worker/internal/datatransfer"
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+const (
+	dataTransferTargetName                     = "data transfer"
+	DefaultDataTransferMetricsScrapingInterval = 60
+)
+
+var defaultDataTransferMetricsConfiguration = models.ComponentMetricsConfiguration{Interval: DefaultDataTransferMetricsScrapingInterval}
+
+type DataTransferMetrics struct {
+	latestConfig atomic.Value
+	daemon       MetricsDaemon
+}
+
+func NewDataTransferMetrics(daemon MetricsDaemon) *DataTransferMetrics {
+	return &DataTransferMetrics{
+		daemon: daemon,
+	}
+}
+
+func (dt *DataTransferMetrics) String() string {
+	return "data transfer metrics"
+}
+
+func (dt *DataTransferMetrics) Init(config models.DeviceConfigurationMessage) error {
+	return dt.Update(config)
+}
+
+func (dt *DataTransferMetrics) Update(config models.DeviceConfigurationMessage) error {
+	newConfiguration := retrieveConfigurationOrDefault(config.Configuration)
+	latestConfig := dt.latestConfig.Load()
+	if latestConfig != nil {
+		oldConfiguration, ok := latestConfig.(*models.ComponentMetricsConfiguration)
+		if !ok {
+			return fmt.Errorf("invalid type %+v", latestConfig)
+		}
+		if oldConfiguration != nil && reflect.DeepEqual(newConfiguration, *oldConfiguration) {
+			// Same configuration, no updates required.
+			return nil
+		}
+	}
```

Comment:
Lines 43 to  53 could be extracted to a method `checkForChanges`

Reply:
will check